### PR TITLE
Plus last update leftovers + some vanilla templates additions

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:

leftover content from plus. All templates has been adopted to include new sniper patrol stuff
CASdive is still WIP
Changes to templates from plus      

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:
WIP
CASdive needs testing
and sniper patrol stuff needs testing

### CODE REVIEW

File: A3A/addons/ultimate/config/plane/cfgPlaneLoadouts.hpp [hunk 1]
```diff
@@ -6,9 +6,20 @@ class planeLoadouts
...
+    {
+       class B_Plane_CAS_01_dynamicLoadout_F
+        {
+            loadout[] = {"","","","","PylonMissile_1Rnd_Bomb_04_F","PylonMissile_1Rnd_BombCluster_03_F","","","",""};
+            mainGun = "Gatling_30mm_Plane_CAS_01_F";
+            bombRacks[] = {"Bomb_03_Plane_CAS_02_F", "BombCluster_02_F"};
+            diveParams[] = {1200, 600, 180, 55, 15, {0,0}};
+        };
+    };
+
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/plane/cfgPlaneLoadouts.hpp]:
```suggestion
    class CASDIVE
    {
        class B_Plane_CAS_01_dynamicLoadout_F
        {
            loadout[] = {"","","","","PylonMissile_1Rnd_Bomb_04_F","PylonMissile_1Rnd_BombCluster_03_F","","","",""};
            mainGun = "Gatling_30mm_Plane_CAS_01_F";
            bombRacks[] = {"Bomb_04_Plane_CAS_01_F", "BombCluster_03_F"};
            diveParams[] = {1200, 600, 180, 55, 15, {0,0}};
        };
    };
```

**Comment by wersal454** [comment]
Ufff. This is gonna be long.

Changelog:
*Addonvehicles - added CIV planes(before they were in civ helis)
   updated civilian example with civ helis and planes, also added variants and animations
   Added sniper patrol pair for every template.(only templates that were in plus use cloaked gear)
   sniper patrol pair consist of patrolspotter and patrolsniper.
   In faction example added cloak version of gear(for snipers) - cloakUniforms,cloakVests,cloakRifles, etc.
*Some small changes to templates:
   additional vanilla helis to civ template,armored orca for CSAT 
   NATO's tracked APC is now IFV(because it's called an IFV lol)
   Vanilla and Aegis templates now use stealth variants of jets dlc planes as ASF
   Vanilla and Aegis templates now use jets dlc planes as CAS + F-35 in aegis used in CAS and in ASF(stealth variant)
   Now all vanilla rivals have helicopter
   Fixes and additions for vanilla rebels (rebels, now can by AAF tanks from tanks dlc + AAF small helicopter(unarmed)).
*CASdive: GIANT list of fixes for regular CAS plane loadouts
   support for CASdive for every plane.
  +Additional planes and their loadouts for clone wars, PLA, RACS+SLA,AMF,Pedagne
   CAS will now pop flares upon exiting it's gun run + it will use it's radar during the run(dunno if it will help AI or not).
   Vanilla and Aegis templates now use stealth variants of jets dlc planes as ASF
   Vanilla and Aegis templates now use jets dlc planes as CAS + F-35 in aegis used in CAS and in ASF(stealth variant).
*one small bug fix fn_createAIcontrols.
*Some small changes for VEHAIRDROP from plus (it will be used more... or less I dunno)
  Airdrop now uses groupsTierAirborne insted of groupsTierMedium.
  Stringtables (in eng and ru) for Sumava map.
*And some other changes from plus I probably forgot about.

**Comment by stutpip123** [comment]
"Vanilla and Aegis templates now use jets dlc planes as CAS + F-35 in aegis used in CAS and in ASF(stealth variant)
Now all vanilla rivals have helicopter"
DLC should only be used if activated in the setup menu for vanilla templates

"+Additional planes and their loadouts for clone wars, PLA, RACS+SLA,AMF,Pedagne"
Those clone wars planes were not added because they physically don't fit into hangars and therefore cannot be used in templates


hard to say if theres any issues without looking through all the code. its a lot of changes for one PR

**Comment by Silence-112** [comment]
Sounds like a perfect case for the unstable branch/test mod

**Comment by stutpip123** [comment]
not until the DLC thing is sorted out. Vanilla templates cannot REQUIRE DLC for anything. It's fine to use the system to check loaded DLC and append vehicles then, but cannot simply replace the vanilla fighter with DLC and have that as the only option

**Comment by wersal454** [comment]
> "Vanilla and Aegis templates now use jets dlc planes as CAS + F-35 in aegis used in CAS and in ASF(stealth variant) Now all vanilla rivals have helicopter" DLC should only be used if activated in the setup menu for vanilla templates
> 
> "+Additional planes and their loadouts for clone wars, PLA, RACS+SLA,AMF,Pedagne" Those clone wars planes were not added because they physically don't fit into hangars and therefore cannot be used in templates
> 
> hard to say if theres any issues without looking through all the code. its a lot of changes for one PR

1. I have not added the big ships/planes 2. Have you looked at plane from TFC? lol. It doesn't fit in vanilla hangar. It immediately blows up.

**Comment by stutpip123** [comment]
Well thats Silences fault. I didn't port TFC

**Comment by wersal454** [comment]
> not until the DLC thing is sorted out. Vanilla templates cannot REQUIRE DLC for anything. It's fine to use the system to check loaded DLC and append vehicles then, but cannot simply replace the vanilla fighter with DLC and have that as the only option

Yeah probably something should be done about that, but that is how vanilla templates always were so ¯\_(ツ)_/¯

**Comment by stutpip123** [comment]
https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/blob/66b6997ddd24aae97e674a90a80332d267289661/A3A/addons/core/Templates/Templates/BWA3/BWA3_AI_BW_Temperate.sqf
See the BWMod templates for how to implement optional content to arrays such as for PlanesAA

**Comment by Silence-112** [comment]
> Well thats Silences fault. I didn't port TFC

We have used TFC multiple times, surprised we didn't figure it out sooner

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf [hunk 2]
```diff
@@ -35,22 +37,22 @@ private _armedVehicles = ["B_T_MRAP_01_gmg_F", "B_T_MRAP_01_hmg_F"];
...
 ["vehiclesAA", ["B_T_APC_Tracked_01_AA_F"]] call _fnc_saveToTemplate;
 
 ["vehiclesTransportBoats", ["B_T_Boat_Transport_01_F"]] call _fnc_saveToTemplate;
 ["vehiclesGunBoats", ["B_T_Boat_Armed_01_minigun_F"]] call _fnc_saveToTemplate;
 
-["vehiclesPlanesCAS", ["B_Plane_CAS_01_dynamicLoadout_F"]] call _fnc_saveToTemplate;
-["vehiclesPlanesAA", ["B_Plane_Fighter_01_F"]] call _fnc_saveToTemplate;
-["vehiclesPlanesTransport", ["B_VTOL_01_infantry_F"]] call _fnc_saveToTemplate;
+private _planeCAS = ["B_Plane_CAS_01_dynamicLoadout_F"];
+private _planeAA = [];
```
Comment (stutpip123) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf]: does this mean if Jets is not loaded then they dont get an AA plane? what happened to the pure vanilla plane?

Comment (wersal454) [reply A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf]: "B_Plane_Fighter_01_F" never had a AA loadout, can add it tho

**Comment by Silence-112** [comment]
I'll merge this if you're fine with it @wersal454

**Comment by wersal454** [comment]
> I'll merge this if you're fine with it @wersal454

hold up, finishing touches

**Comment by Silence-112** [comment]
Is this good yet or no?

**Comment by wersal454** [comment]
> Is this good yet or no?

no, finishing vanilla rivals

**Comment by Silence-112** [comment]
np, just comment when it is

**Comment by wersal454** [comment]
Changelog №2:
AAF:
AAF will now use only only theirs 230mm artillery
AAF tracked APC now is IFV
CSAT:
CSAT's tracked APC is also moved to IFV category
no transport plane without APEX dlc enabled
some fixes to helicopters categorisation
probably mentioned that already but CSAT will now use ION's armored orca with CSAT camo
CSAT will now use AAF's 230mm artillery with CSAT camo
CSAT will now have railgun tank from the latest update
LDF:
basicly same as above.
NATO:
Added mover variant of the truck to basic vehicles, since it now has 1 cargo slot
Added AAF MRAP with NATO camo variant
Same template changes as above + If apex dlc is enabled transport helis will now have CTRG variant of vanilla NATO transport helicopter
Mortar APC from WS was added to the artillery
Civ:
Many dlc checks were added to vehicles + some missing vehicles were added
Civis will now wear costumes from Contact and Arts of war dlcs
Reb:
Many dlc checks were added to vehicles
by default FIA & LL rebels will now don't have planes until APEX is enabled (may break some shit, same as missing looting heli)
Now rebels can by civ versions of  AAF heli and CSAT heli
some missing vehicles were added
WS armored offroads were moved from blackmarket to regular store
On the blackmarket now you will be able to by: 
small uavs from dlc's (for example deming UGV + IDAPs antimine uav)
Static designators from marksmen dlc
automatic HMG and GMG
Armed and unarmed vanilla UGVs
AAF's light tanks from tanks dlc
3 new helicopters (2 armed ones)
Tons of new clothing + Art of war dlc backpacks
new heads
camera as a starting equipment(binocular) if you have WS.
Riv:
checks for dlc were redone.
now if you have APEX dlc enabled Exegermenos will now have both vanilla offroad and APEX one
uav's from Laws of war and WS were added to rivals arsenal
If you have WS enabled rivals will have light CSAT APC from WS(armed and unarmed)
If you have Tanks dlc enabled rivals will now have autocannon wiesel from the dlc
Rivals will now use more gear(not weapons) with and without dlc enabled

Special forces units will now sometimes wear CBRN suits if you enable the Contact dlc
No new weapons from dlc were added (no time) something for the future update
Tons of camo variants
probably mentioned that already, artillery can now fire cluster and mines munition (if I understand that correctly) or they will now fire 32 rounds of HE ammo and then switch to cluster and etc.
Enemy templates will now use vanilla planes as AA without JETS dlc enabled 
Tons of checks for dlc's(for example no light tanks until you enable the dlc)
Templates will now use police vans from Laws of war dlc
No more radar and sam until you enable the Jets dlc
Many additions to camo variants of the enemy vehicles
Officers will now wear officer uniforms from Art of war dlc if you enable that dlc
Sometimes you will spot enemies wearing gasmask from contact dlc
Enemy squad leaders will now have IR grenades in their inventory
Elite squads can now spawn with shield backpack from WS cdls (no clue if they will use it or not)

**Comment by Silence-112** [comment]
That is a chonky changelog indeed
Good to go yet?

**Comment by wersal454** [comment]
yes

**Pull request merged**